### ISSUE TITLE
[win] SegFault

### ISSUE THREAD

**Issue by sequencerr** [oven-sh/bun#11538]
### Relevant log output

```shell
============================================================
Bun Canary v1.1.12-canary.1 (43f0913c) Windows x64
Args: "bun" "run" ".\src\index.js"
Features: jsc abort_signal(5) dotenv fetch(3) http_server https_server transpiler_cache(6) tsconfig(10)
Builtins: "bun:main" "node:assert" "node:async_hooks" "node:buffer" "node:child_process" 
"node:crypto" "node:events" "node:fs" "node:fs/promises" "node:http" "node:https" "node:net" "node:os" "node:path" "node:querystring" "node:stream" "node:string_decoder" "node:tls" "node:tty" "node:url" "node:util" "node:util/types" "node:zlib" "node:punycode"       
Elapsed: 10426805ms | User: 10390ms | Sys: 5468ms
RSS: 0.33GB | Peak: 0.33GB | Commit: 0.69GB | Faults: 1058069

panic(main thread): Segmentation fault at address 0x0
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

 https://bun.report/1.1.12/wr243f0913AipijgFA2AA
```

### Stack Trace (bun.report)

Bun v1.1.12 ([`43f0913`](<https://github.com/oven-sh/bun/tree/43f0913c3875cbf260c60868268adc5953b09a80>)) on windows x86_64 [RunCommand]

Segmentation fault at address 0x00000000

**Comment by sequencerr** [comment]
randomly crashed `nodemon --exec "bun run .\src\index.js" --watch src --watch ../client/build --no-update-notifier`. can't really share code, had express.js, @ngrok/ngrok.

**Comment by sequencerr** [comment]
============================================================
Bun Canary v1.1.12-canary.1 (43f0913c) Windows x64
Args: "bun" "run" ".\src\index.js"
Features: jsc dotenv fetch(3) http_server https_server transpiler_cache(6) tsconfig(10)  
Builtins: "bun:main" "node:assert" "node:async_hooks" "node:buffer" "node:child_process" 
"node:crypto" "node:events" "node:fs" "node:fs/promises" "node:http" "node:https" "node:net" "node:os" "node:path" "node:querystring" "node:stream" "node:string_decoder" "node:tls" "node:tty" "node:url" "node:util" "node:util/types" "node:zlib" "node:punycode"       
Elapsed: 122675ms | User: 1765ms | Sys: 703ms
RSS: 0.29GB | Peak: 0.30GB | Commit: 0.39GB | Faults: 91130

panic(main thread): cast causes pointer to be null
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

 https://bun.report/1.1.12/wr243f0913AioijgF4zroLw/vweu0zxeg0wxew+33G24sh2C6+hyF802yF49s2Fqi4BA0eNpLTiwuUUhOLC1OLVYoyM/MK0ktUijJV0hKVcgrzckBAK96CzY

[nodemon] app crashed - waiting for file changes before starting...

got another one. crashed second time not while after. but bun report link is kind of complete

**Comment by nektro** [comment]
this might fix but hard to be sure with no reproduction

```diff
diff --git a/src/io/PipeReader.zig b/src/io/PipeReader.zig
index f6c6855cf..7ce663c64 100644
--- a/src/io/PipeReader.zig
+++ b/src/io/PipeReader.zig
@@ -1014,11 +1014,11 @@ pub const WindowsBufferedReader = struct {
         this.flags.is_paused = false;
         const source: Source = this.source orelse return .{ .err = bun.sys.Error.fromCode(bun.sys.E.BADF, .read) };
         bun.assert(!source.isClosed());
+        source.setData(this);
 
         switch (source) {
             .file => |file| {
                 file.fs.deinit();
-                source.setData(this);
                 const buf = this.getReadBufferWithStableMemoryAddress(64 * 1024);
                 file.iov = uv.uv_buf_t.init(buf);
                 if (uv.uv_fs_read(uv.Loop.get(), &file.fs, file.file, @ptrCast(&file.iov), 1, if (this.flags.use_pread) @intCast(this._offset) else -1, onFileRead).toError(.write)) |err| {
```

**Comment by sequencerr** [comment]
I could try testing when I have free time.

**Comment by cirospaciari** [comment]
Looks like this was fixed a while ago, please open a new issue with a repro if still encountering this issue after v1.3.1.
